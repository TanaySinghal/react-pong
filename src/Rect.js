import styled from 'styled-components';

const defaultWidth = 30;
const defaultHeight = 250;
const defaultColor = '#000';
const defaultX = 0;
const defaultY = 0;

const Rect = styled.div.attrs({
  style: ({ x = defaultX, y = defaultY }) => ({
    transform: `translate(${x}px,${y}px)`
  })
})`
  background-color: ${({ color = defaultColor }) => color};
  width: ${({ width = defaultWidth }) => width}px;
  height: ${({ height = defaultHeight }) => height}px;
  display: flex;
  transition: transform 150ms ease-in-out;
`;

export default Rect;
