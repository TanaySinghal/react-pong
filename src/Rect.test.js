import React from 'react';
import { shallow } from 'enzyme';
import 'jest-styled-components';
import Rect from './Rect';

const defaultWidth = '30px';
const defaultHeight = '250px';
const defaultBgColor = '#000';
const defaultX = 0;
const defaultY = 0;

test('should render without crashing', () => {
  const wrapper = shallow(<Rect />);
  expect(wrapper).toMatchSnapshot();
});

test(`dimensions should default to ${defaultWidth} X ${defaultHeight} if they are not provided`, () => {
  const wrapper = shallow(<Rect />);

  expect(wrapper).toHaveStyleRule('width', defaultWidth);
  expect(wrapper).toHaveStyleRule('height', defaultHeight);
});

test(`dimensions should equal the dimensions provided`, () => {
  const width = '300px';
  const height = '100px';
  const wrapper = shallow(<Rect width={width} height={height} />);

  expect(wrapper).toHaveStyleRule('width', width);
  expect(wrapper).toHaveStyleRule('height', height);
});

test(`background-color should default to ${defaultBgColor} if not provided`, () => {
  const wrapper = shallow(<Rect />);
  expect(wrapper).toHaveStyleRule('background-color', defaultBgColor);
});

test(`background-color should equal the bgColor provided`, () => {
  const bgColor = '#fff';
  const wrapper = shallow(<Rect bgColor={bgColor} />);

  expect(wrapper).toHaveStyleRule('background-color', bgColor);
});

test(`coordinates should default to (${defaultX}, ${defaultY}) if not provided`, () => {
  const wrapper = shallow(<Rect />);

  expect(wrapper).toHaveStyleRule(
    'transform',
    `translate(${defaultX}px, ${defaultY}px)`
  );
});

test(`coordinates should be equal to the values provided`, () => {
  const x = 300;
  const y = 20;
  const wrapper = shallow(<Rect x={x} y={y} />);

  expect(wrapper).toHaveStyleRule('transform', `translate(${x}px, ${y}px)`);
});
