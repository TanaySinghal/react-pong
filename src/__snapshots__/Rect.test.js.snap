// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render without crashing 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <styled.div />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "className": "sc-bdVaJa jQFNVU",
      "style": Object {
        "transform": "translate(0px,0px)",
      },
    },
    "ref": null,
    "rendered": null,
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "className": "sc-bdVaJa jQFNVU",
        "style": Object {
          "transform": "translate(0px,0px)",
        },
      },
      "ref": null,
      "rendered": null,
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
